ingress-nginx-controller-1.12.yaml

```diff
@@ -571,6 +571,82 @@ subpackages:
+    pipeline:
+      - working-directory: images/custom-error-pages/rootfs/
+        pipeline:
+          - runs: |
+              mkdir -p ${{targets.contextdir}}/usr/bin
+              mkdir -p /go/src/k8s.io/ingress-nginx/images/custom-error-pages
+              cp -r * /go/src/k8s.io/ingress-nginx/images/custom-error-pages/
+              cd /go/src/k8s.io/ingress-nginx/images/custom-error-pages
+              go get . && \
+              go build -o ${{targets.contextdir}}/usr/bin/nginx-errors .
+    test:
+      environment:
+        contents:
+          packages:
+            - wait-for-it
+            - curl
+            - ingress-nginx-custom-error-pages-compat-${{vars.nginx-ingress-major-minor}}
+      pipeline:
+        - runs: |
+            /usr/bin/nginx-errors &
+            wait-for-it -t 10 localhost:8080
+
+            # check healthz endpoint make sure it returns 200
+            http_code=$(curl -s -o /dev/null -w "%{http_code}" http://localhost:8080/healthz)
+
+            if [ "$http_code" != "200" ]; then
+              echo "Healthz endpoint returned $http_code"
+              exit 1
+            fi
+
+            # I already checked that the following error exists in the upstream custom error pages image
+            # unexpected error reading return code: strconv.Atoi: parsing "": invalid syntax. Using 404
+            # serving custom error response for code 404 and format text/html from file /www/404.html
+            http_code=$(curl -s -o /dev/null -w "%{http_code}" http://localhost:8080/)
+
+            if [ "$http_code" != "404" ]; then
+              echo "Custom error page did not return 404"
+              exit 1
+            fi
+
+            # also check the message
+            message=$(curl -s http://localhost:8080/)
+
+            # Trim message (removes leading/trailing spaces & newlines)
+            message=$(echo "$message" | tr -d '\n' | sed 's/^[ \t]*//;s/[ \t]*$//')
+
+            if [[ "$message" != "<span>The page you're looking for could not be found.</span>" ]]; then
+                echo "Custom error page did not return expected message"
+                exit 1
+            fi
+
+  - name: ingress-nginx-custom-error-pages-compat-${{vars.nginx-ingress-major-minor}}
+    description: OTel plugin for ingress nginx controller ${{vars.nginx-ingress-major-minor}}
+    dependencies:
+      provides:
+        - ingress-nginx-custom-error-pages-compat=${{package.full-version}}
+    pipeline:
+      - working-directory: images/custom-error-pages/rootfs/
+        pipeline:
+          - runs: |
+              mkdir -p ${{targets.contextdir}}/www
+              mkdir -p ${{targets.contextdir}}/etc/
+              cp -r www/* ${{targets.contextdir}}/www/
+              cp -r etc/* ${{targets.contextdir}}/etc/
+              ln -sf /usr/bin/nginx-errors ${{targets.contextdir}}/nginx-errors
+    test:
+      pipeline:
+        - runs: |
+            ls -lah /www
+            ls -lah /etc
```

Comment:
Are there specific files in here to check for?

Reply:
yep thank you!